File: Models/ViewModels/PostVM.cs
using GoWheels_WebAPI.Models.Entities;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class PostVM : BaseModelVM
    {
        public required string Name { get; set; }
        public required string Image { get; set; }
        public List<PostImageVM> Images { get; set; } = new List<PostImageVM>();
        public string? Description { get; set; }
        public required int Seat { get; set; }
        public string? RentLocation { get; set; }
        public bool HasDriver { get; set; }
        public required decimal Price { get; set; }
        public bool Gear { get; set; }
        public string? Fuel { get; set; }
        public decimal FuelConsumed { get; set; }
        public int RideNumber { get; set; }
        public float AvgRating { get; set; }
        public bool IsAvailable { get; set; }
        public bool IsDisabled { get; set; }
        public string? CarTypeName { get; set; }
        public string? CompanyName { get; set; }
        public UserVM User { get; set; } = null!;
        public List<PostAmenityVM> PostAmenities { get; set; } = new List<PostAmenityVM>();
        public List<PostPromotionVM> PostPromotions { get; set; } = new List<PostPromotionVM>();
        public List<RatingVM> Ratings { get; set; } = new List<RatingVM>();
    }
}

File: FE/src/Model/PostVM.js
// src/models/PostVM.js

import PostImageVM from './PostImageVM';
import UserVM from './UserVM';
import PostAmenityVM from './PostAmenityVM';
import PostPromotionVM from './PostPromotionVM';
import RatingVM from './RatingVM';
import BaseModel from './BaseModelVM';
export default class PostVM extends BaseModel{
  constructor(data = {}) {
    super(data.id || null, data.createdById, data.createdOn, data.modifiedById, data.modifiedOn, data.isDeleted);
    this.name = data.name || '';
    this.image = data.image || '';
    this.images = (data.images || []).map(img => new PostImageVM(img));
    this.description = data.description || '';
    this.seat = data.seat || 0;
    this.rentLocation = data.rentLocation || '';
    this.hasDriver = data.hasDriver || false;
    this.pricePerHour = data.pricePerHour || 0;
    this.pricePerDay = data.pricePerDay || 0;
    this.gear = data.gear || false;
    this.fuel = data.fuel || '';
    this.fuelConsumed = data.fuelConsumed || 0;
    this.rideNumber = data.rideNumber || 0;
    this.avgRating = data.avgRating || 0;
    this.isAvailable = data.isAvailable || false;
    this.isDisabled = data.isDisabled || false;
    this.carTypeId = data.carTypeId || 0;
    this.carTypeName = data.carTypeName || '';
    this.companyId = data.companyId || 0;
    this.companyName = data.companyName || '';
    this.user = new UserVM(data.user || {});
    this.postAmenities = (data.postAmenities || []).map(amenity => new PostAmenityVM(amenity));
    this.postPromotions = (data.postPromotions   || []).map(promo => new PostPromotionVM(promo));
    this.ratings = (data.ratings || []).map(rating => new RatingVM(rating));
  }
}
